From fd46364c7ca512923884068905f19655b0379ae2 Mon Sep 17 00:00:00 2001
From: Chase Douglas <chase.douglas@canonical.com>
Date: Fri, 24 Feb 2012 15:05:51 -0800
Subject: [PATCH 0013/1767] UBUNTU: SAUCE: (no-up) Input: synapticss - Set
 buttonpad property for all clickpads

no-up comments: Keep both, so long as they continue to apply cleanly.
The patches apply only to a couple of old Dell minis, and Dell has said
they don't intend to use those touchpads again. Upstreaming these
patches stalled due to lack of information/response, and continuing to
pursue it probably isn't worth the effort, so they can be marked no-up.
There's no harm in keeping the patches, but if they become a problem
they can be dropped.

Signed-off-by: Chase Douglas <chase.douglas@canonical.com>
Acked-by: Seth Forshee <seth.forshee@canonical.com>
Acked-by: Andy Whitcroft <andy.whitcroft@canonical.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 drivers/input/mouse/synaptics.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/input/mouse/synaptics.c b/drivers/input/mouse/synaptics.c
index 6025eb4..e678212 100644
--- a/drivers/input/mouse/synaptics.c
+++ b/drivers/input/mouse/synaptics.c
@@ -1250,7 +1250,9 @@ static void set_input_params(struct psmouse *psmouse,
 		/* Clickpads report only left button */
 		__clear_bit(BTN_RIGHT, dev->keybit);
 		__clear_bit(BTN_MIDDLE, dev->keybit);
-	}
+	} else if (SYN_CAP_CLICKPAD2BTN(priv->ext_cap_0c) ||
+		   SYN_CAP_CLICKPAD2BTN2(priv->ext_cap_0c))
+		__set_bit(INPUT_PROP_BUTTONPAD, dev->propbit);
 }
 
 static ssize_t synaptics_show_disable_gesture(struct psmouse *psmouse,
-- 
1.9.3

